### DISCUSSION TITLE
Can I sort by a field in the work item

### DISCUSSION THREAD

**Discussion by MarkGTI** [rfennell/AzurePipelines#1274] [Q&A]
Hi All

We have a field called "Release Notes Section" which is used to group release notes in to logical areas (different from Epics and features)

Is there a way to sort and group the released notes based on this, e.g

**AR**
Bug ix .....
New feature ....

**AP**
Bug fix 1....
New feature 2...
New feature 3....

**Comment by rfennell** [comment]
I think you will have to write a custom extension that takes the complete array of WI does the filtering/grouping

Have a look at [this sample](https://github.com/rfennell/AzurePipelines/blob/main/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%203/each_with_sort_by_field.js) as a starting point

**Reply by MarkGTI** [reply]
Thanks for quick reply. If tried that example but I get this error

_Error Processing handlebars [Error: Missing helper: "each_with_sort_by_field"]_

I'm guessing I put helper in wrong place: here's my template. Does the helper need to go somewhere else?

# Notes for release  {{releaseDetails.releaseDefinition.name}}    
**Release Number**  : {{releaseDetails.name}}
**Release completed** : {{releaseDetails.modifiedOn}}     
**Build Number**: {{buildDetails.id}}
**Compared Release Number**  : {{compareReleaseDetails.name}}    
**Build Trigger PR Number**: {{lookup buildDetails.triggerInfo 'pr.number'}} 

# Builds with associated WI/CS ({{builds.length}})
{{#forEach builds}}
{{#if isFirst}}## Builds {{/if}}
-  Build {{this.build.buildNumber}}
{{/forEach}}

# Work Items ({{workItems.length}})
{{#each_with_sort_by_field  this.workItems "Custom.ReleaseNotesSection"}}
  * **{{lookup this.fields 'Custom.ReleaseNotesSection'}}
   - **{{this.id}}**  {{lookup this.fields 'Custom.ReleaseNotesTitle'}}
   - **Description** {{{lookup this.fields 'Custom.ReleaseNotes'}}}
{{/each_with_sort_by_field}}


const handlebars = require("handlebars");
module.exports = {each_with_sort_by_field(array, key, opts) {
    if(opts.data) {
        data  = handlebars.createFrame(opts.data) 
    }

    // the sort
    array = array.sort (function (a, b) {
        a = a.fields[key]
        b = b.fields[key]
        if (a > b) return 1 
        if (a == b) return 0
        if (a < b) return -1
    });

    // the iterator
    var len = array.length;
    var buffer = '';
    var i = -1;
  
    while (++i < len) {
      var item = array[i];
      data.index = i;
      item.index = i + 1;
      item.total = len;
      item.isFirst = i === 0;
      item.isLast = i === (len - 1);
      buffer += opts.fn(item, {data: data});
    }
    return buffer;
}}
